### ISSUE TITLE
Attach Schema objects to the producer subscription object

### ISSUE THREAD

**Issue by eleduardo** [homeaway/stream-registry#56]
Desired Behavior
==============================

Attach AVRO Schema objects to the producer (and possibly consumer) object(s) so when a subscription is requested the metadata for the records is readily available in a single request. 

Benefits
==============================

While in the java world it makes little sense to send down the schema(s) associated with the stream since POJOs already serialize/deserialize objects with the encoding, other stacks like Python, NodeJS and even golang are required to produce data... this is very AVRO specific but it could apply to other formats in the future.

With the current REST API it is still achievable to do this for the non-java stacks but it requires a secondary request to the server to fetch the actual stream metadata making the protocol a little chattier. If the subscription response had the schema(s) objects attached to them (just like the stream object does) then there would be no need to fetch that.

For reference:

Golang AVRO example
https://github.com/linkedin/goavro (notice the need to pass the schema)

Python Confluent AVRO producer
https://github.com/confluentinc/confluent-kafka-python

**Comment by prabhakar1983** [comment]
if I understand correctly, the ask is , the JSON response of Consumer Registration and producer Registration should also return the Schemas along with the bootstrapUris

**Comment by cricket007** [comment]
On the Avro side, I would like to see something like [Apache Hive](https://cwiki.apache.org/confluence/display/Hive/AvroSerDe) 

- `'avro.schema.literal'` = The JSON encoded payload
- `'avro.schema.url'` = An HTTP link to an AVSC file

**Comment by neoword** [comment]
@eleduardo , @codeburke: ALL **_non-generic_** clients, irrespective of language, SHALL REQUIRE the _specific version_ they depend on to function.  Therefore, they _are_ build time dependencies.  Granted that non-typed languages like Node and Python won't require generated libraries (they can just use JSON or Dictionaries), however the SERDE still requires the _**specific version**_ that was used (e.g. expected) at BUILD TIME.  If you provide a different version at runtime, the avro-resolution may(will) result in erratic results.  

@cricket007, an example of a _generic_ consumer is that registers with Apache Hive (to your comment above)

* To obviate the need to pass specific version numbers around, I recommend this be handled in the yet-to-be-published stream modeling pipeline that will handle this for the various languages
```
java --> `import foo.bar.Type`
node --> `require "TypeSchema"`
python --> `import "TypeSchema"`
```

Suggesting we mark this as WONT-DO and will be handled when we open source stream modelling pipeline:

TL;DR:
- producer already knows its schema, is required at build time
- generic-consumer doesn't care (its provided the schema as part of message delivery)
- non-generic-consumer already knows its schema, as it is required at build time (so that proper AVRO resolution takes place and to ensure the consumer doesn't break - see [AvroDeserializer in apache avro source code](https://avro.apache.org/docs/1.8.2/api/java/org/apache/avro/hadoop/io/AvroDeserializer.html#AvroDeserializer(org.apache.avro.Schema,%20org.apache.avro.Schema,%20org.apache.avro.io.DatumReader)).
